PWGHF/TableProducer/candidateCreatorXicToXiPiPi.cxx

```diff
@@ -0,0 +1,655 @@
+
+      //----------------------------fit SV and create XicPlus track------------------
+      auto trackParCovCharmBachelor0 = getTrackParCov(trackCharmBachelor0);
+      auto trackParCovCharmBachelor1 = getTrackParCov(trackCharmBachelor1);
+
+      // reconstruct the 3-prong secondary vertex
+      if (df.process(trackCasc, trackParCovCharmBachelor0, trackParCovCharmBachelor1) == 0) {
+        continue;
+      }
+
+      //----------------------------calculate physical properties-----------------------
+      // Charge of charm baryon
+      int signXic = casc.sign() < 0 ? +1 : -1;
+
+      // get SV properties
+      const auto& secondaryVertex = df.getPCACandidate();
+      auto chi2SV = df.getChi2AtPCACandidate();
+      auto covMatrixSV = df.calcPCACovMatrixFlat();
+
+      // get track momenta
+      trackParCovCharmBachelor0 = df.getTrack(1);
+      trackParCovCharmBachelor1 = df.getTrack(2);
+      trackCasc = df.getTrack(2);
+      std::array<float, 3> pVecXi;
+      std::array<float, 3> pVecPi0;
+      std::array<float, 3> pVecPi1;
+      trackCasc.getPxPyPzGlo(pVecXi);
+      trackParCovCharmBachelor0.getPxPyPzGlo(pVecPi0);
+      trackParCovCharmBachelor1.getPxPyPzGlo(pVecPi1);
+
+      // get invariant mass of Xic candidate
+      auto arrayMomenta = std::array{pVecXi, pVecPi0, pVecPi1};
+      massXiPiPi = RecoDecay::m(std::move(arrayMomenta), std::array{massXiMinusFromPdg, massPionFromPdg, massPionFromPdg});
+
+      // get track impact parameters
+      // This modifies track momenta!
+      auto primaryVertex = getPrimaryVertex(collision);
+      auto covMatrixPV = primaryVertex.getCov();
+      // calculate impact parameter
+      o2::dataformats::DCA impactParameterCasc;
+      o2::dataformats::DCA impactParameter0;
+      o2::dataformats::DCA impactParameter1;
+      trackCasc.propagateToDCA(primaryVertex, bz, &impactParameterCasc);
+      trackParCovCharmBachelor0.propagateToDCA(primaryVertex, bz, &impactParameter0);
+      trackParCovCharmBachelor1.propagateToDCA(primaryVertex, bz, &impactParameter1);
+
+      // calculate cosine of pointing angle
+      std::array<float, 3> pvCoord = {collision.posX(), collision.posY(), collision.posZ()};
+      double cpaLambda = casc.v0cosPA(collision.posX(), collision.posY(), collision.posZ());
+      double cpaXYLambda = RecoDecay::cpaXY(pvCoord, vertexV0, pVecV0);
+      double cpaXi = casc.casccosPA(collision.posX(), collision.posY(), collision.posZ());
+      double cpaXYXi = RecoDecay::cpaXY(pvCoord, vertexCasc, pVecCasc);
+
+      // get invariant mass of Xi-pi pairs
+      auto arrayMomentaXiPi0 = std::array{pVecXi, pVecPi0};
+      massXiPi0 = RecoDecay::m(std::move(arrayMomentaXiPi0), std::array{massXiMinusFromPdg, massPionFromPdg});
+      auto arrayMomentaXiPi1 = std::array{pVecXi, pVecPi1};
+      massXiPi1 = RecoDecay::m(std::move(arrayMomentaXiPi1), std::array{massXiMinusFromPdg, massPionFromPdg});
+
+      // get uncertainty of the decay length
+      double phi, theta;
```

Comment:
where are they defined?

Reply:
Do you mean the variables to store the reconstructed masses? At the beginning of the struct, outside of the process functions.

Reply:
my bad, I was overlooking that their definition is in `getPointDirection`